src/main/scala/sample/reactivekafka/KafkaReaderCoordinator.scala

```diff
@@ -41,19 +41,53 @@ class KafkaReaderCoordinator(mat: Materializer, topicName: String) extends Actor
       .kafkaOffsetsStorage()
       .commitInterval(1200 milliseconds)
     )
+
     log.debug("Starting the reader")
-    Source(consumerWithOffsetSink.publisher)
-      .map(processMessage)
+    val currencyRateSource = Source(consumerWithOffsetSink.publisher)
+    val currencyBroadcaster = Source(currencyRateSource.runWith(Sink.publisher(true)))
+
+    val alertsFlow = Flow[MessageAndMetadata[Array[Byte], CurrencyRateUpdated]]
+      .map(processAlert)
+      .filter(alert => alert != None)
+      .map(alert => alert.get)
+
+
+    // commit offset so we don't process the kafka message again
+    currencyBroadcaster
       .withAttributes(ActorAttributes.supervisionStrategy(processingDecider))
-      .to(consumerWithOffsetSink.offsetCommitSink).run()
```

Comment:
Correct me if I'm wrong.
So to guarantee once only delivery you would have to acknowledge that you have received the message.
This is irrespective of whether you have processed it or now.

The intention was to acknowledge the message after broadcast so that any subsequent streams can process the message once.

Is there are better way of doing this?